### PR TITLE
Handle invalid error

### PR DESCRIPTION
**Pull request by sota1235**
I think fujiwara doesn't allow all Error. 
He must cry with invalid exception.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+    if ( opts.ignoreSpaces ) {
+      opts.ignoreChars = opts.ignoreChars.concat([' ', '　']);
+    }
+    return text.split('').map((char) => {
+      if ( char.charCodeAt() < 0x20 ) return char;
+      if ( opts.ignoreChars.indexOf(char) !== -1 ) return char;
+      return char + opts.separator;
+    }).join('');
+  } catch (err) {
+    const errorMessage = 'そ゛ん゛な゛ひ゛き゛す゛う゛わ゛か゛ら゛ね゛え゛よ゛お゛お゛お゛お゛！゛！゛！゛';
```
Comment (hoto17296) [inline index.js:8]: Good message!!!

**Comment by sota1235** [comment]
I missed the way to handle Error...I'll fix.

File: index.js [hunk 2]
```diff
@@ -1,16 +1,23 @@
+const FujiwaraError = require('./fujiwara-error');
```
Comment (hoto17296) [inline index.js:3]: `const` declaration needs `"use strict";` mode in Node.js v4.

This module supports Node.js (>= 4).

File: test/index.js [hunk 3]
```diff
@@ -27,4 +28,16 @@ describe('fujiwara', () => {
     assert.equal(fujiwara(text), expect);
   });
 
+  it('should despair with invalid parameter', () => {
+    const invalidCases = [
```
Comment (hoto17296) [inline test/index.js:34]: `"use strict";`

File: fujiwara-error.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+function FujiwaraError(message) {
```
Comment (hoto17296) [inline fujiwara-error.js]: ES2015 class syntax is available in Nodejs (>= 4).

``` js
"use strict";

class FujiwaraError extends Error {
  ...
}
```

**Review by hoto17296 — CHANGES_REQUESTED** [review]
🙏

File: index.js (hunk 2 above)
Comment (sota1235) [reply index.js:3]: Oops, I'll use `var`:bow:

**Comment by sota1235** [comment]
@hoto17296 Please review again 🐶

**Review by hoto17296 — APPROVED** [review]
Cooool 🎉 🎉 🎉 
Thanks for your pull request!!

**Pull request merged**